### PR TITLE
update BluetoothStack.wprp with additional providers

### PR DESCRIPTION
**Pull request by SamPavezaMS**

### CODE REVIEW

File: bluetooth/tracing/BluetoothStack.wprp [hunk 1]
```diff
@@ -489,22 +489,27 @@
...
+        <Keyword Value="0x7fffffff"/>
+      </Keywords>
+    </EventProvider>
     <EventProvider Id="Microsoft.Bluetooth.Audio" Name="6F3A682E-24E1-4073-93F2-15327F542536" Level="4" NonPagedMemory="false">
       <Keywords>
         <Keyword Value="0x7fffffff"/>
       </Keywords>
     </EventProvider>
-    <EventProvider Id="Microsoft.Bluetooth.Audio.Verbose" Name="6F3A682E-24E1-4073-93F2-15327F542536" Level="5" NonPagedMemory="false">
+    <EventProvider Id="ApxGuid.Verbose" Name="96760AC7-C416-43A7-AA7F-1A676942589C" Level="4" NonPagedMemory="false">
```
Comment (erikpe-msft) [inline bluetooth/tracing/BluetoothStack.wprp:502]: >4 [](http://example.com/codeflow?start=91&length=1)

This should probably be "5" and "ApxGuid" should be "4"

File: bluetooth/tracing/BluetoothStack.wprp [hunk 2]
```diff
@@ -526,6 +531,31 @@
...
+      <Keywords>
+        <Keyword Value="0x7fffffff"/>
+      </Keywords>
+    </EventProvider>
+    <EventProvider Id="AcxLogger.Verbose" Name="2393C85D-B36A-4FE2-86A9-12500A05B76A" Level="5" NonPagedMemory="true">
+      <Keywords>
+        <Keyword Value="0x7fffffff"/>
+      </Keywords>
+    </EventProvider>
+    <EventProvider Id="AcxLogger" Name="2393C85D-B36A-4FE2-86A9-12500A05B76A" Level="5" NonPagedMemory="true">
```
Comment (erikpe-msft) [inline bluetooth/tracing/BluetoothStack.wprp:544]: >5 [](http://example.com/codeflow?start=85&length=1)

Intentionally "5" in non-Verbose?

File: bluetooth/tracing/BluetoothStack.wprp [hunk 3]
```diff
@@ -526,6 +531,31 @@
...
+      <Keywords>
+        <Keyword Value="0x7fffffff"/>
+      </Keywords>
+    </EventProvider>
+    <EventProvider Id="ApxKm.Verbose" Name="E3A25276-CD6E-40DC-8BB6-1659A5E00952" Level="4" NonPagedMemory="true">
```
Comment (erikpe-msft) [inline bluetooth/tracing/BluetoothStack.wprp:549]: >"4" [](http://example.com/codeflow?start=88&length=3)

Swap level with "ApxKm"?

File: bluetooth/tracing/BluetoothStack.wprp [hunk 4]
```diff
@@ -881,7 +914,6 @@
             <EventProviderId Value="Microsoft.Windows.Apps.PhoneShellUi"/>
             <EventProviderId Value="Microsoft.Windows.Apps.PhoneUtilsProviders"/>
             <EventProviderId Value="Microsoft.Windows.Apps.TelephonyInteractiveUser"/>
-            <EventProviderId Value="Microsoft.Windows.TestExecution.WexLogger"/>
```
Comment (erikpe-msft) [inline bluetooth/tracing/BluetoothStack.wprp:884]: ><EventProviderId Value="Microsoft.Windows.TestExecution.WexLogger"/> [](http://example.com/codeflow?start=12&length=68)

Please add this back

File: bluetooth/tracing/BluetoothStack.wprp [hunk 5]
```diff
@@ -1043,7 +1081,6 @@
...
```
Comment (erikpe-msft) [inline bluetooth/tracing/BluetoothStack.wprp:1046]: ><EventProviderId Value="Microsoft.Windows.TestExecution.WexLogger"/> [](http://example.com/codeflow?start=12&length=68)

Please add this back

File: bluetooth/tracing/BluetoothStack.wprp [hunk 6]
```diff
@@ -1092,6 +1129,11 @@
             <EventProviderId Value="Microsoft.Windows.Bluetooth.Telemetry.Proximity"/>
             <EventProviderId Value="Microsoft.Windows.Bluetooth.Telemetry.QuickPair"/>
             <EventProviderId Value="Microsoft.Windows.Bluetooth.Telemetry.Telephony.AsyncPhoneController"/>
+            <EventProviderId Value="Microsoft.Windows.Audio.PortClass"/>
+            <EventProviderId Value="portcls"/>
+            <EventProviderId Value="Microsoft.Windows.Audio.AudioClassExtension"/>
+            <EventProviderId Value="AcxLogger"/>
+            <EventProviderId Value="ApxKm"/>
```
Comment (erikpe-msft) [inline bluetooth/tracing/BluetoothStack.wprp:1136]: >ApxKm [](http://example.com/codeflow?start=36&length=5)

.Verbose?

File: bluetooth/tracing/BluetoothStack.wprp [hunk 7]
```diff
@@ -1092,6 +1129,11 @@
             <EventProviderId Value="Microsoft.Windows.Bluetooth.Telemetry.Proximity"/>
             <EventProviderId Value="Microsoft.Windows.Bluetooth.Telemetry.QuickPair"/>
             <EventProviderId Value="Microsoft.Windows.Bluetooth.Telemetry.Telephony.AsyncPhoneController"/>
+            <EventProviderId Value="Microsoft.Windows.Audio.PortClass"/>
+            <EventProviderId Value="portcls"/>
+            <EventProviderId Value="Microsoft.Windows.Audio.AudioClassExtension"/>
+            <EventProviderId Value="AcxLogger"/>
```
Comment (erikpe-msft) [inline bluetooth/tracing/BluetoothStack.wprp:1135]: >AcxLogger [](http://example.com/codeflow?start=36&length=9)

.Verbose?

**Review by erikpe-msft — CHANGES_REQUESTED** [review]
:clock1:

**Comment by erikpe-msft** [comment]
In general, how would you describe the verbosity of these new providers, when compared with providers that we already have? We do have to be sensitive to significantly increasing the size of our trace files, especially if it can impact "all" Bluetooth trace scenarios.

**Comment by SamPavezaMS** [comment]
> In general, how would you describe the verbosity of these new providers, when compared with providers that we already have? We do have to be sensitive to significantly increasing the size of our trace files, especially if it can impact "all" Bluetooth trace scenarios.

@Erik not sure how to contextualize this, maybe we should meet offline. After pairing I see 71 apx traces and 23K acx traces.

File: bluetooth/tracing/BluetoothStack.wprp [hunk 8]
```diff
@@ -1043,7 +1067,7 @@
             <EventProviderId Value="Microsoft.Windows.Apps.PhoneShellUi"/>
             <EventProviderId Value="Microsoft.Windows.Apps.PhoneUtilsProviders"/>
             <EventProviderId Value="Microsoft.Windows.Apps.TelephonyInteractiveUser"/>
-            <EventProviderId Value="Microsoft.Windows.TestExecution.WexLogger"/>
+		<EventProviderId Value="Microsoft.Windows.TestExecution.WexLogger"/>
```
Comment (erikpe-msft) [inline bluetooth/tracing/BluetoothStack.wprp:1070]: >		 [](http://example.com/codeflow?start=0&length=2)

two wild tabs appear :)

File: bluetooth/tracing/BluetoothStack.wprp [hunk 9]
```diff
@@ -875,13 +893,13 @@
             <EventProviderId Value="Microsoft-Windows-Telephony-PhoneUtil"/>
             <EventProviderId Value="Microsoft-Windows-Telephony-PhoneService"/>
             <EventProviderId Value="Microsoft-Windows-Telephony-TelephonyInteractiveUser"/>
+		<EventProviderId Value="Microsoft.Windows.TestExecution.WexLogger"/>
```
Comment (erikpe-msft) [inline bluetooth/tracing/BluetoothStack.wprp:896]: >		 [](http://example.com/codeflow?start=0&length=2)

oh no, more wild tabs!

**Review by erikpe-msft — APPROVED** [review]
:shipit:

File: bluetooth/tracing/BluetoothStack.wprp (hunk 8 above)
Comment (SamPavezaMS) [reply bluetooth/tracing/BluetoothStack.wprp]: added back

**Pull request merged**